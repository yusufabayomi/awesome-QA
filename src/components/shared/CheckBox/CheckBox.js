const CheckBox = ({ label, input }) => {
  return (
    <>
      <label className='form-label'>
        <input {...input} type='checkbox' /> {label}
      </label>
    </>
  );
};

export default CheckBox;
