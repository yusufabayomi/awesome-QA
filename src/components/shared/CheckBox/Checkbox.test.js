import React from 'react';
import { shallow } from 'enzyme';
import CheckBox from './CheckBox';

describe('Renders CheckBox Component', () => {
  let wrapper;
  beforeEach(() => {
    const props = {
      label: 'Delay entry by 5 seconds',
    };
    wrapper = shallow(<CheckBox {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has a label', () => {
    expect(wrapper.find('label')).toHaveLength(1);
  });

  it('has an input field', () => {
    expect(wrapper.find('input')).toHaveLength(1);
  });

  it('label value must be Delay entry by 5 seconds', () => {
    expect(wrapper.find('label').text()).toBe('Delay entry by 5 seconds');
  });
});
