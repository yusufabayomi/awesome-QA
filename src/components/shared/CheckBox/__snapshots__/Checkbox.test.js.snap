// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CheckBox Component renders correctly 1`] = `
<Fragment>
  <label
    className="form-label"
  >
    <input
      type="checkbox"
    />
     
    Delay entry by 5 seconds
  </label>
</Fragment>
`;
